packit/config/package_config.py

```diff
@@ -254,6 +254,151 @@ def __eq__(self, other: object):
+                "upstream_tag_template": package_config.upstream_tag_template,
+                "dist_git_base_url": package_config.dist_git_base_url,
+                "dist_git_namespace": package_config.dist_git_namespace,
+                "spec_source_id": package_config.spec_source_id,
+                "create_pr": package_config.create_pr,
+                "actions": package_config.actions,
+            }
+        )
+
+    @staticmethod
+    def validate_list(package_config: PackageConfig) -> dict:
+        from packit.schema import PackageConfigSchema
+
+        return PackageConfigSchema().validate(
+            {
+                "create_tarball_command": package_config.create_tarball_command,
+                "current_version_command": package_config.current_version_command,
+                "allowed_gpg_keys": package_config.allowed_gpg_keys,
+                "patch_generation_ignore_paths": package_config.patch_generation_ignore_paths,
+            }
+        )
+
+
+class PackageConfigValidation:
+    def __init__(self, package_config: PackageConfig):
+        self.config_file_path: Optional[str] = package_config.config_file_path
+        self.schema_primitive_errors: str = str(
+            PrimitiveValidationError(package_config)
+        )
+        self.schema_list_errors: str = str(ListValidationError(package_config))
+
+    def get_validation_errors_str(self) -> str:
+        if not (self.schema_primitive_errors or self.schema_list_errors):
+            return f"{self.config_file_path} is valid and ready to be used"
+
+        validation_errors = f"{self.config_file_path} does not pass validation:\n"
+        validation_errors += self.schema_primitive_errors
+        validation_errors += self.schema_list_errors
+
+        return validation_errors
+
+    def __repr__(self) -> str:
+        return self.get_validation_errors_str()
+
+
+class ValidationError:
+    default_error_messages = {
+        "Field may not be null.": "{} has incorrect type, it may not be {}\n",
+        "Not a valid string.": "{} has incorrect type, it should be str, is {} right now\n",
+        "Not a valid utf-8 string.": "{} has incorrect type, it should be utf-8 str, is {} right now\n",
+        "Not a valid list.": "{} has incorrect type, it should be list, is {} right now\n",
+        "Not a valid boolean.": "{} has incorrect type, it should be boolean, is {} right now\n",
+    }
+
+    def __init__(
+        self, error: str, field_name: str, field_value, field_category: str = "* field"
+    ):
+        self.error: str = error
+        self.field_name: str = f"{field_category} {field_name}"
+        self.field_value = field_value
+
+    def __repr__(self) -> str:
+        error = self.default_error_messages.get(self.error)
+        if not error:
+            return f"* {self.field_name}: {error}"
+        return error.format(self.field_name, type(self.field_value).__name__)
+
+
+class PrimitiveValidationError:
```

Comment:
I am a bit confused about the naming. Classes that end with `Error` or `Exceptions` are usually errors or exceptions. (And can be used in the `try-except`.)